java-checks/src/main/java/org/sonar/java/checks/NullPointerCheck.java

```diff
@@ -409,6 +414,32 @@ private void visitRelationalNotEqualTo(BinaryExpressionTree tree) {
     }
   }
 
+  class AssignmentVisitor extends BaseTreeVisitor {
+    Set<VariableSymbol> assignedSymbols = new HashSet<>();
+
+    public Set<VariableSymbol> findAssignedVariables(Tree tree) {
+      tree.accept(this);
+      return assignedSymbols;
+    }
+
+    public Set<VariableSymbol> findAssignedVariables(List<? extends Tree> trees) {
+      for (Tree tree : trees) {
+        tree.accept(this);
+      }
+      return assignedSymbols;
+    }
+
+    @Override
+    public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+      // currently we only handle assignment of the form: identifier =
+      // TODO(merciesa): array[expr] = and mse.identifier = requires advanced tracking and are left outside of the scope
```

Comment:
mse.identifier is not a local variable so it is outside of scope. 
However array[index] = ... ; can easily be tracked. But for me this is not relevant for what we are currently doing : array should be initialized alltogether or array access will raise NPE. 
Therefore, I think this comment is not relevant here.